### PR TITLE
Implement checked operations on Bool

### PR DESCRIPTION
**Pull request by nalimilan**
This fixes the current breakage on master. Cf https://github.com/JuliaLang/julia/pull/15227#issuecomment-190329492 and https://github.com/JuliaLang/julia/pull/15228.

@eschnett This is a very mechanical change, do you think it's a good way to do this?

### CODE REVIEW

File: base/checked.jl [hunk 1]
```diff
@@ -37,7 +38,7 @@ checked_mod{T<:Integer}(x::T, y::T) = no_op_err("checked_mod", T)
 checked_cld{T<:Integer}(x::T, y::T) = no_op_err("checked_cld", T)
 
 typealias SignedInt Union{Int8,Int16,Int32,Int64,Int128}
-typealias UnsignedInt Union{UInt8,UInt16,UInt32,UInt64,UInt128}
+typealias UnsignedInt Union{UInt8,UInt16,UInt32,UInt64,UInt128, Bool}
```
Comment (nalimilan) [inline base/checked.jl]: I'll fix this spacing issue (and the one below).

**Comment by nalimilan** [comment]
My first attempt did not follow the types returned by unchecked operations (e.g. `true + true === 1`). Here's a new version which adds separate tests.

**Comment by tkelman** [comment]
I'm going to merge this for now to fix CI. @eschnett please do review, if you have any comments we can refine going forward.

File: base/checked.jl [hunk 2]
```diff
@@ -12,6 +12,7 @@ import Core.Intrinsics: box, unbox,
        checked_srem_int,
        checked_uadd_int, checked_usub_int, checked_umul_int, checked_udiv_int,
        checked_urem_int
+import Base: no_op_err
```
Comment (eschnett) [inline base/checked.jl]: Why do you need this import? It seems unused.

Comment (tkelman) [inline base/checked.jl]: https://github.com/JuliaLang/julia/blob/58bdac3aa431eb67b8809bdad8f56c32c115de06/base/checked.jl#L29-L38 ?

**Comment by nalimilan** [comment]
@eschnett BTW, I wonder why these lines do not check the actual value at the same time as the type. They are redundant with tests below, which use `===` and therefore check the type, right?
https://github.com/JuliaLang/julia/blob/58bdac3aa431eb67b8809bdad8f56c32c115de06/test/checked.jl#L11-L22

**Comment by eschnett** [comment]
Yes, the tests below check both value and type. This is probably for historic reasons, and could be cleaned up.

**Comment by nalimilan** [comment]
OK, see https://github.com/JuliaLang/julia/pull/15304

**Pull request merged**